java/compaction/compaction-core/src/main/java/sleeper/compaction/strategy/DelegatingCompactionStrategy.java

```diff
@@ -57,58 +59,46 @@ public DelegatingCompactionStrategy(
     }
 
     @Override
-    public void init(InstanceProperties instanceProperties, TableProperties tableProperties) {
+    public void init(InstanceProperties instanceProperties, TableProperties tableProperties, List<FileReference> fileReferences, List<Partition> partitions) {
         CompactionJobFactory factory = new CompactionJobFactory(instanceProperties, tableProperties);
         leafStrategy.init(instanceProperties, tableProperties, factory);
         shouldCreateJobsStrategy.init(instanceProperties, tableProperties);
-        table = tableProperties.getStatus();
+        this.table = tableProperties.getStatus();
+        this.leafPartitionIds = partitions.stream()
+                .filter(Partition::isLeafPartition)
+                .map(Partition::getId)
+                .collect(Collectors.toList());
+        this.filesWithJobIdByPartitionId = fileReferences.stream()
+                .filter(file -> file.getJobId() != null)
+                .collect(Collectors.groupingBy(FileReference::getPartitionId));
+        this.filesWithNoJobIdByPartitionId = fileReferences.stream()
+                .filter(file -> file.getJobId() == null)
+                .sorted(Comparator.comparingLong(FileReference::getNumberOfRecords))
+                .collect(Collectors.groupingBy(FileReference::getPartitionId));
```

Comment:
Please can we put this in some sort of index class, rather than make it part of the compaction strategy?